docs/guide/web-simulation.md

```diff
@@ -324,6 +324,71 @@ This folder also contains a `server.sh` utility script to automatically start an
 
 Please make sure that the `WEBOTS_HOME` variable is set before running the simulation and session server scripts.
 
+#### Using Docker
+
+The simulation server can also be run in a Docker container.
+In this section we provide a sample Docker setup that could be used to run the simulation on localhost.
+
+You can use the following `Dockerfile` to build your Docker image.
+```
+FROM cyberbotics/webots:R2021a-ubuntu18.04
+
+RUN apt update
+RUN apt install -y nano psmisc firejail python3-pip
+RUN pip3 install websocket-client tornado nvidia-ml-py psutil requests distro
+RUN pip3 install -U pynvml --user
+ENV AUDIODEV=null
+ENV DEBIAN_FRONTEND=noninteractive
+ENV DISPLAY=:99
+ENV LIBGL_ALWAYS_SOFTWARE=true
+ENV WEBOTS_HOME=/usr/local/webots
+ENV QTWEBENGINE_DISABLE_SANDBOX=1
+RUN echo 'alias python=python3' >> ~/.bashrc
+COPY server/config /usr/local/webots/resources/web/server/config
+COPY server/key /usr/local/webots/resources/web/server/key
+COPY server.sh /usr/local/server.sh
+RUN chmod 654 /usr/local/server.sh
+CMD ["/usr/local/server.sh", ""]
+```
+
+To correctly setup and automatically run the simulation server, you should provide the following files:
+* `server/config`: a folder containing the simulation and session configuration files.
+    The current example `server.sh` the local configuration by default, i.e. you should provide the `server/config/session/local.json` and `server/config/simulation/local.json` files or specify the configuration files to be used.
+    For example, `server/config/session/localhost.js`
+    ```
+    {
+      "port": 1999,
+      "server": "localhost",
+      "simulationServers": [
+        "localhost:2000"
+      ]
+    }
+    ```
+    and `server/config/simulation/localhost.js`
+    ```
+    {
+      "port": 2000
+    }
+    ```
+* `server/key`: a folder containing your website host keys needed for validation (see [Session server](#session-server) section).
+* `server.sh`: a script that configures the virtual screen and starts the simulation and session servers.
+    ```bash
+    #!/bin/sh
+    Xvfb :99 -screen 0 1024x768x16 &
+    cd $WEBOTS_HOME/resources/web/server
+    ./server.sh start
+    while true; do sleep 1000; done
```

Comment:
That seems odd... What is the purpose of that? Not to kill the server.sh script by exiting? Couldn't we do something like this instead?
```suggestion
    nohup ./server.sh start
```

Reply:
The purpose is to keep the Docker container running otherwise it will return and stop immediately after starting the server script. 
Not sure that nohup will work. 

There is another alternatively that is to add a run option to run directly the container in background.
But it has the drawback that requires more steps to stop it so I personally prefer this one for testing (in any case it is not a system that could be used as-is in production). 

I will improve it on Monday.

Reply:
Docker needs a foreground running process to continue running, otherwise it stops immediately.
That is why `nohup` doesn't work and even running the container in background doesn't make any difference.
We could avoid this loop by using a custom `server.sh` script that doesn't run the session and/or simulation server in background.

Checking again the documentation of Docker, it seems that is designed to run a single server the container, so another possibility is to run one container for the session server and one for the simulation server.

Reply:
It looks like it is overkill to run one Docker just for the session server.